src/relayserver/ReputationManager.ts

```diff
@@ -0,0 +1,107 @@
+import { Address } from '../relayclient/types/Aliases'
+import { LoggerInterface } from '../common/LoggerInterface'
+import { ReputationStoreManager } from './ReputationStoreManager'
+import { ReputationChange } from './ReputationEntry'
+
+export interface ReputationManagerConfiguration {
+  /** All new Paymasters start with their reputation set to this value  */
+  initialReputation: number
+  /** No matter how good the Paymaster is, it's reputation cannot go above this value */
+  maximumReputation: number
+  /** If the reputation is below this value, transactions will only be allowed each {@link throttleDelayMs} */
+  throttleReputation: number
+  /** Minimum interval between transactions paid by a specific throttled Paymaster */
+  throttleDelayMs: number
+  /** Paymasters with reputation below this value will never be served */
+  blockReputation: number
+  /** If a paymaster loses this number of reputation points within {@link abuseTimeWindowMs}, it will be blocked for {@link abuseBlockDurationMs} */
+  abuseReputationChange: number
+  abuseTimeWindowMs: number
+  abuseBlockDurationMs: number
+  /** After {@link abuseBlockDurationMs}, the Paymaster reputation will be reset to this value */
+  abuseTimeoutReputation: number
+}
+
+const defaultReputationConfig: ReputationManagerConfiguration = {
+  initialReputation: 3,
+  maximumReputation: 100,
+  throttleReputation: 5,
+  throttleDelayMs: 60000,
+  blockReputation: 0,
+  abuseTimeoutReputation: 1,
+  abuseReputationChange: 20,
+  abuseTimeWindowMs: 3600000,
```

Comment:
its great idea to put the type into the parameter name.
I still think better unit is "Min". Since its javascript, fractions are possible so in tests you can do `abuseTimeWindowMin:1/600` for 0.1sec....